esp-hal-common/src/gpio.rs

```diff
@@ -634,6 +647,17 @@ macro_rules! impl_input {
                 }
                 self
             }
+
+            fn disconnect_input_from_peripheral(&mut self, signal: InputSignal) -> &mut Self {
+                // Reset this GPIO to plain GPIO
+                self.set_alternate_function(AlternateFunction::Function2);
```

Comment:
`Function2` is the GPIO function ONLY on ESP32 - on other chips it's currently Function1 but for future chips it might be even some other function. That's what the macro parameter `$gpio_function` is for